examples/prom/linear_elasticity_global_rom.cpp

```diff
@@ -0,0 +1,497 @@
+    }
+    if (static_cond) {
+        a->EnableStaticCondensation();
+    }
+    a->Assemble();
+
+    HypreParMatrix A;
+    Vector B, X;
+    a->FormLinearSystem(ess_tdof_list, x, *b, A, X, B);
+    if (myid == 0)
+    {
+        cout << "done." << endl;
+        cout << "Size of linear system: " << A.GetGlobalNumRows() << endl;
+    }
+    assembleTimer.Stop();
+
+    // 16. The offline phase
+    if (fom || offline)
+    {
+        // 17. Define and apply a parallel PCG solver for A X = B with the BoomerAMG
+        //     preconditioner from hypre.
+        HypreBoomerAMG* amg = new HypreBoomerAMG(A);
+        if (amg_elast && !a->StaticCondensationIsEnabled())
+        {
+            amg->SetElasticityOptions(fespace);
+        }
+        else
+        {
+            amg->SetSystemsOptions(dim, reorder_space);
+        }
+        HyprePCG* pcg = new HyprePCG(A);
+        pcg->SetTol(1e-8);
+        pcg->SetMaxIter(500);
+        pcg->SetPrintLevel(2);
+        pcg->SetPreconditioner(*amg);
+        solveTimer.Start();
+        pcg->Mult(B, X);
+        solveTimer.Stop();
+        delete amg;
+        delete pcg;
+
+        // 18. take and write snapshot for ROM
+        if (offline)
+        {
+            bool addSample = generator->takeSample(X.GetData(), 0.0, 0.01);
+            generator->writeSnapshot();
+            delete generator;
+            delete options;
+        }
+    }
+
+    // 19. The online phase
+    if (online) {
+        // 20. read the reduced basis
+        assembleTimer.Start();
+        CAROM::BasisReader reader(basisName);
+        spatialbasis = reader.getSpatialBasis(0.0);
+        numRowRB = spatialbasis->numRows();
+        numColumnRB = spatialbasis->numColumns();
+        if (myid == 0) printf("spatial basis dimension is %d x %d\n", numRowRB,
+                                  numColumnRB);
+
+        // libROM stores the matrix row-wise, so wrapping as a DenseMatrix in MFEM means it is transposed.
+        DenseMatrix* reducedBasisT = new DenseMatrix(spatialbasis->getData(),
+                numColumnRB, numRowRB);
+
+        // 21. form inverse ROM operator
+        Vector abv(numRowRB), bv(numRowRB), bv2(numRowRB);
+        Vector reducedRHS(numColumnRB), reducedSol(numColumnRB);
+        DenseMatrix invReducedA(numColumnRB);
+        for (int j = 0; j < numColumnRB; ++j) {
+            reducedBasisT->GetRow(j, bv);
+            A.Mult(bv, abv);
+            reducedRHS(j) = bv * B;
+            for (int i = 0; i < numColumnRB; ++i) {
+                reducedBasisT->GetRow(i, bv2);
+                invReducedA(i, j) = abv * bv2;
+            }
+        }
+        invReducedA.Invert();
+        assembleTimer.Stop();
+
+        // 22. solve ROM
+        solveTimer.Start();
+        invReducedA.Mult(reducedRHS, reducedSol);
+        solveTimer.Stop();
+
+        // 23. reconstruct FOM state
+        reducedBasisT->MultTranspose(reducedSol, X);
+        delete reducedBasisT;
```

Comment:
In the online phase, or in a post-processing phase, it would be nice to compute the error of the ROM solution (lifted to FOM space) with respect to a FOM solution stored in a file. That is more complicated and I suppose not necessary for this PR, but usually that is desired to verify the ROM is accurate.

Reply:
@dylan-copeland @chldkdtn I agree that it'd be nice to have and I already did the function to calculate the relative error, but in a standalone script. I can adapt it to be used in the online phase instead.

Reply:
If it helps, in lines 889-901 of this other example, the relative error is also computed: https://github.com/LLNL/libROM/blob/master/examples/prom/dg_advection_global_rom.cpp

Reply:
Hi! I have now implemented the relative error calculation. Feel free to check it out @dylan-copeland @chldkdtn @kevinhkhuynh